src/components/forms/Checkbox.jsx

```diff
@@ -11,62 +11,89 @@ import CheckIndeterminateIcon from 'lib/icons/react/CheckIndeterminateIcon';
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const Checkbox = ({
-  showLabel = true,
-  indeterminate = false,
-  disabled = false,
-  label,
-  inputRef,
-  ...rest
-}) => {
-  const id = uuidv4();
+const Checkbox = React.forwardRef(
+  ({ showLabel = true, disabled = false, label, defaultChecked, ...rest }, ref) => {
+    const [checkedState, setChecked] = useState(rest.checked || defaultChecked || false);
+    const checked = rest.checked != null ? rest.checked : checkedState;
 
-  const IconUnchecked = CheckEmptyIcon;
-  const IconChecked = indeterminate ? CheckIndeterminateIcon : CheckFilledIcon;
+    // We should maybe fire onChange in "addition" to setChecked, based on whether we
+    // interpret checkbox as being "controlled" or not
+    const onChange = rest.onChange ? rest.onChange : (e) => setChecked(e.target.checked);
```

Comment:
as in, the user always controls the state?

I mean the only problem I see with that, is it could be very useful if we wanted to use more patterns that have us just grabbing the resulting form data when the user clicks submit. That being said, we probably couldn't easily do that if we're doing any sort of form validation (e.g. react hook form)? I guess that's maybe a better question.

Also a side note: we need to come up language that distinguished a controlled component / uncontrolled component, vs a controlled / uncontrolled input (not just for us but for cbi-site, as I've already ran into confusion on this with other developers)

Reply:
I go by the React definition, which is that when react state is responsible for setting the value of an input, it is a controlled component: 
https://reactjs.org/docs/uncontrolled-components.html

I don't have a strong opinion on controlled vs uncontrolled inputs, although React recommends using controlled inputs.

The main issue I see with the current way `Checkbox` works in this PR is that it manages its own checked state until you pass an `onChange`. Once I pass `onChange`, I am fully responsible for managing the `checked` value. If that was intended, we should make sure all form components behave the same way.

Reply:
At this point I'm really really leaning towards leaving this out of v8. I'm sort of lost as to what we're even trying to achieve with this PR anymore, and it's probably going to take more effort to align here.